runtime/near-vm-logic/src/array_utils.rs

```diff
@@ -0,0 +1,63 @@
+//! Polyfils for missing const-generic array APIs in the std.
+
+use std::slice::ChunksExact;
+
+/// Splits `&[u8; A + B]` into `(&[u8; A], &[u8; B])`.
+pub(crate) fn split_array<const N: usize, const L: usize, const R: usize>(
+    xs: &[u8; N],
+) -> (&[u8; L], &[u8; R]) {
```

Comment:
That said I see why it was done the way it is.

Reply:
>especially since returning a (&[u8; L], &[u8; N-L]) is not possible.

Returning `(&[u8; L], &[u8; N-L])` is exactly what is happening here, albeit with an ugly API (an extra const parameter). See the static assert inside